packages/opentelemetry-node/package.json

```diff
@@ -64,6 +65,7 @@
     "@opentelemetry/context-async-hooks": "^0.6.1",
     "@opentelemetry/core": "^0.6.1",
     "@opentelemetry/tracing": "^0.6.1",
+    "deepmerge": "^4.2.2",
```

Comment:
Definitely do not want to depend on some new random dependency for such simple functionality

Reply:
do we want deep merge or shallow merge?

Reply:
seems healthy fwiw https://www.npmjs.com/package/deepmerge

Reply:
+1 to not adding a dependency. I think a shallow merge is fine here, and then we can rely on each plugin handling its own config merge if needed.

Reply:
I also think shallow merge is fine. The "default" configuration is just an empty configuration for all plugins anyways so there is no need to merge with it. Default settings are handled within the plugin itself.

Reply:
sgtm, but what I still do not understand, if we just shallow merge, with regards to default settings being handled within the pluging itself if:

```
        plugins: {
          'http': {
            ignoreUrls: ['foo'],
          },
        }
```

would result in http being enabled and it's path being `'@opentelemetry/plugin-http' ` or not?

Reply:
No you will need to make the `path` property required. Or you will need to implement the config merge, but no need to make it a true deep merge. One level should be enough.

Reply:
gotcha!

Reply:
Alright 2 level merge working, I've added a todo to automatically enable unless explicitly disabled, user configured non-default plugins, but that could be a subsequent PR.

Reply:
pardon the forced pushes, should be done now.